result_test.go

```diff
@@ -409,11 +409,13 @@ func TestToMapBy(t *testing.T) {
 
 func TestToSlice(t *testing.T) {
```

Comment:
can we have tests for edge cases:

1. output is nil slice.
2. cap(out)>len(result): we get the same slice, just resized.
3. cap(out)==len(result): we get the same slice, just resized.
4. cap(out)<len(result): we get a new slice with len(out)=len(result) and cap(out)==??
5. len(out)>0 and appending/growing doesn't lose the original elements in out.